app/models/order.rb

```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
 
   validates :name, :address, :email, presence: true
 
+  scope :by_date, Proc.new {|from, to| where created_at: (from||Date.today)..(to||Date.today) }
```

Comment:
- spacing issues 🤦🏻 
- Why are we using `Proc.new` format here?
- Can we use default parameters?
- whats the datatype of `created_at`? What datatype should the default parameters be?
- `user.orders.by_date​ ​should​ ​work​ ​and​ ​it​ ​should​ ​return​ ​orders​ ​placed​ ​by​ ​user
between​ ​given​ ​dates.​ ​Check​ ​the​ ​query​ ​difference​ ​in​ ​this​ ​one​ ​and​ ​the​ ​above​ ​one`
  - Add a text file where you run the queries, and explain the difference

Reply:
[order_by_date.txt](https://github.com/Garvshah01/Depot/files/11280125/order_by_date.txt)

Reply:
done